DebugMenu/DebugMenu/Wrappers/DebugToggle.swift

```diff
@@ -23,9 +24,13 @@ public struct DebugToggle: DynamicProperty  {
             storage.value
         }
         nonmutating set {
-            storage.value = newValue
-            didSet?(newValue)
-            publisher.send(newValue)
+            Task { @MainActor in
```

Comment:
Do we need to run this on the main thread? It makes me a little nervous blocking it unless we have to, and I feel like there could be situations you want to run something in the background before settings?

Reply:
**TL;DR**
I removed the `@MainActor` annotation and it seems like it's working just fine. Which is surprising to me but I guess because the `DebugToggle` conforms to `DynamicProperty` it makes sure that the code runs on the main actor.

**Long version**
AFAIK the `@MainActor` doesn't block the main thread when there is another async task called in a different context. I just thought that in most use cases there will be shown some UI elements and we should make sure it's executed on the main thread.

I did a quick test by delaying the alert presentation by 10 seconds.

![image](https://github.com/user-attachments/assets/6315e742-d091-48a4-a64c-ea329b330f22)

The app is still responsive and the code continues as expected after the 10 seconds. But as I mentioned at the top it's not necessary.

https://github.com/user-attachments/assets/8d88b804-46b5-469b-9e4d-83d6ac8615a5